docker/ci/build

```diff
@@ -15,23 +16,37 @@ function build_and_test_image() {
   : ===
   : === build and test $container_name
   : ===
-  bundle exec puppet-docker build "$container_name" --repository pcr-internal.puppet.net/release-engineering
+  bundle exec puppet-docker build "$container_name" --no-cache --repository pcr-internal.puppet.net/release-engineering --version "$version" --build-arg postgres_version="$postgres_version" --no-latest
   bundle exec puppet-docker spec "$container_name"
 }
 
 function push_image() {
   container_name="$1"
+  container_version="$2"
   : ===
   : === push $container_name
   : ===
-  bundle exec puppet-docker push "$container_name" --repository pcr-internal.puppet.net/release-engineering
+  bundle exec puppet-docker push "$container_name" --repository pcr-internal.puppet.net/release-engineering --version "$container_version" --no-latest
 }
 
 : ===
 : === bundle install to get ready
 : ===
 bundle install --path "$BUNDLER_PATH"
 
+: ===
+: === If we do not git pull --unshallow we get really weird results with git describe
```

Comment:
Yeah, I imagine you might be able to get away with a clone --depth, but guessing it'd have to be "deep enough" to include the most recent tag, and even if that worked, offhand I'm not sure if you could determine it cheaply (or simply) enough to be worth it.

Reply:
Yeah, I don't think there's a reliable-enough way to do this, so I'd rather stick with `--unshallow`